packages/migrate/src/commands/DbExecute.ts

```diff
@@ -0,0 +1,171 @@
+  ${chalk.dim('$')} prisma db execute --preview-feature --file ./script.sql --schema schema.prisma
+
+  Execute the SQL script from stdin to the database url.
+  ${chalk.dim(
+    '$',
+  )} echo 'SELECT \`hello world\`;' | prisma db execute --preview-feature --stdin --url="mysql://localhost/testdb"
+
+`)
+
+  public async parse(argv: string[]): Promise<string | Error> {
+    const args = arg(
+      argv,
+      {
+        '--help': Boolean,
+        '-h': '--help',
+        '--stdin': Boolean,
+        '--file': String,
+        '--schema': String,
+        '--url': String,
+        '--preview-feature': Boolean,
+        '--telemetry-information': String,
+      },
+      false,
+    )
+
+    if (isError(args)) {
+      return this.help(args.message)
+    }
+
+    if (args['--help']) {
+      return this.help()
+    }
+
+    if (!args['--preview-feature']) {
+      throw new DbExecuteNeedsPreviewFeatureFlagError()
+    }
+
+    // One of --stdin or --file is required
+    if (args['--stdin'] && args['--file']) {
+      throw new Error(
+        `--stdin and --file cannot be used at the same time. Only 1 must be provided. 
+See \`${chalk.green(getCommandWithExecutor('prisma db execute -h'))}\``,
+      )
+    } else if (!args['--stdin'] && !args['--file']) {
+      throw new Error(
+        `Either --stdin or --file must be provided.
+See \`${chalk.green(getCommandWithExecutor('prisma db execute -h'))}\``,
+      )
+    }
+
+    // One of --url or --schema is required
+    if (args['--url'] && args['--schema']) {
+      throw new Error(
+        `--url and --schema cannot be used at the same time. Only 1 must be provided.
+See \`${chalk.green(getCommandWithExecutor('prisma db execute -h'))}\``,
+      )
+    } else if (!args['--url'] && !args['--schema']) {
+      throw new Error(
+        `Either --url or --schema must be provided.
+See \`${chalk.green(getCommandWithExecutor('prisma db execute -h'))}\``,
+      )
+    }
+
+    let script = ''
+    // Read file
+    if (args['--file']) {
+      script = fs.readFileSync(path.resolve(args['--file']), 'utf-8')
+    }
+    // Read stdin
+    if (args['--stdin']) {
+      script = await getStdin()
+      // If input is empty, stop here
+      if (!script) {
+        throw new Error(
+          `--stdin was passed but the input was empty. See \`${chalk.green(
+            getCommandWithExecutor('prisma db execute -h'),
+          )}\``,
+        )
+      }
+    }
+
+    let datasourceType: EngineArgs.DbExecuteDatasourceType
+
+    const schemaPath = await getSchemaPath(args['--schema'])
+    if (!schemaPath) {
+      throw new NoSchemaFoundError()
+    }
+
+    // Execute command(s) to url passed
+    if (args['--url']) {
+      datasourceType = {
+        tag: 'url',
+        url: args['--url'],
+      }
+    } else {
+      // Execute command(s) to url from schema
+      datasourceType = {
+        tag: 'schema',
+        schema: schemaPath,
+      }
+    }
```

Comment:
I'm not sure I understand the control flow here, but doesn't this `else` means we're defaulting to the URL from the prisma schema in the default location? I don't remember if it's written down in the notion doc, but we had consensus that this was too dangerous — `--url` or `--schema` should always be specified.

Reply:
There are a few conditions before that ensure the params are all passed, see https://github.com/prisma/prisma/pull/11371/files/3265e18552dbc9c588c62cb816a3a6a6a84567a3#diff-cdc4535306baf77165c5d00cadbdab8c9077c5888b3cdd72af6f4b6b8c119b5eR84-R108

Reply:
I think if none or both are present there is an error earlier.